contracts/governance/Vesting/DevelopmentFund.sol

```diff
@@ -0,0 +1,328 @@
+			require(txStatus, "Not enough token sent to change release schedule.");
+		} else if (remainingTokens > _releaseTotalTokenAmount) {
+			/// If there are more tokens than required, send the extra tokens back.
+			bool txStatus = SOV.transfer(msg.sender, remainingTokens.sub(_releaseTotalTokenAmount));
+			require(txStatus, "Token not received by the Locked Owner.");
+		}
+
+		/// Updating the remaining tokens in contract.
+		remainingTokens = _releaseTotalTokenAmount;
+
+		/// Finally we update the token release schedule.
+		releaseDuration = _releaseDuration;
+		releaseTokenAmount = _releaseTokenAmount;
+
+		emit TokenReleaseChanged(msg.sender, _releaseDuration.length);
+	}
+
+	/**
+	 * @notice Transfers all of the remaining tokens in an emergency situation.
+	 * @dev This could be called when governance or development fund might be compromised.
+	 */
+	function transferTokensByUnlockedTokenOwner() public onlyUnlockedTokenOwner {
+		bool txStatus = SOV.transfer(safeVault, remainingTokens);
+		require(txStatus, "Token transfer was not successful. Check receiver address.");
+
+		emit LockedTokenTransferByUnlockedOwner(msg.sender, safeVault, remainingTokens);
+
+		remainingTokens = 0;
+	}
+
+	/**
+	 * @notice Withdraws all unlocked/released token.
+	 * @param _amount The amount to be withdrawn.
+	 */
+	function withdrawTokensByUnlockedTokenOwner(uint256 _amount) public onlyUnlockedTokenOwner {
+		require(_amount > 0, "Zero can't be withdrawn.");
+
+		uint256 count; /// To know how many elements to be removed from the release schedule.
+		uint256 amount = _amount; /// To know the total amount to be transferred.
+		uint256 newLastReleaseTimeMemory = lastReleaseTime; /// Better to use memory than storage.
+		uint256 releaseLength = releaseDuration.length.sub(1); /// Also checks if there are any elements in the release schedule.
+
+		/// Getting the amount of tokens, the number of releases and calculating the total duration.
+		while (amount > 0 && newLastReleaseTimeMemory.add(releaseDuration[releaseLength]) < block.timestamp) {
+			if (amount >= releaseTokenAmount[releaseLength]) {
+				amount = amount.sub(releaseTokenAmount[releaseLength]);
+				newLastReleaseTimeMemory = newLastReleaseTimeMemory.add(releaseDuration[releaseLength]);
+				count++;
+			} else {
+				/// This will be the last case, if correct amount is passed.
+				releaseTokenAmount[releaseLength] = releaseTokenAmount[releaseLength].sub(amount);
+				amount = 0;
+			}
+			releaseLength--;
+		}
+
+		/// Checking to see if atleast a single schedule was reached or not.
+		require(count > 0 || amount == 0, "No release schedule reached.");
+
+		/// If locked token owner tries to send a higher amount that schedule
+		uint256 value = _amount.sub(amount);
+
+		emit UnlockedTokenWithdrawalByUnlockedOwner(msg.sender, value, count);
+
+		/// Now clearing up the release schedule.
+		while (count > 0) {
+			releaseDuration.length--;
```

Comment:
direct array `.length` manipulation is meh.
- you should use `releaseDuration.pop()` instead
- in this particular case you can just shorten arrays without the loop to save gas: `releaseDuration.length -= count`. 
    - in general it might lead to underflow and arbitrary consequences.

Reply:
Used `.pop()`, rectified in commit 06291a1

`releaseDuration.length -= count` here `count` is the number of entries to remove. So, that won't work. And it is in the same lines of reducing the length I think.

Reply:
regarding `-= count`: it zeroes out removed (popped) elements and reduces `length` value. I assume it is the correct behavior.

After Istanbul hardfork gas savings (compared to multi-`pop` approach) become mush lower, about 1000 per `pop`
Overall `pop` is better unless gas saving is the top priority.

Reply:
Understood :+1: Rectified in commit dd2f7d9